Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/messaging/client/CONNECTCXFClientFactory.java

```diff
@@ -52,6 +53,11 @@
     @Override
     public <T> CONNECTClient<T> getCONNECTClientSecured(ServicePortDescriptor<T> portDescriptor,
         AssertionType assertion, String url, String targetHomeCommunityId, String serviceName) {
+
+        if (StringUtils.isEmpty(url)) {
+            throw new IllegalArgumentException("URL for a CONNECT Client must be provided. Got '" + url + "'");
```

Comment:
why is the url add to the string when it empty?

Reply:
Handles both "Null" values and it being empty.